core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -708,13 +713,17 @@ class LogManager(logDirs: Seq[File],
    */
   private def deleteLogs(): Unit = {
     try {
-      while (!logsToBeDeleted.isEmpty) {
-        val (removedLog, scheduleTimeMs) = logsToBeDeleted.take()
+      def hasDeletableLog: Boolean = {
+        if (!logsToBeDeleted.isEmpty) {
+          val (_, scheduleTimeMs) = logsToBeDeleted.peek()
+          scheduleTimeMs + currentDefaultConfig.fileDeleteDelayMs - time.milliseconds() <= 0
```

Comment:
Perhaps we can return the time to wait for the next task and in line 742, wait for that amount instead of always the default currentDefaultConfig.fileDeleteDelayMs. This will make the deletion time of the next item more accurate.

Reply:
@junrao Thanks for the review, I have updated the code.